### PR TITLE
HEL-206 | Streamline add to collection

### PR DESCRIPTION
**Pull request by tuovinensanttu**
Removed cropper from add to collection modal and moved the methods under `AjaxAddToCollection` class.

### CODE REVIEW

File: hkm/urls.py [hunk 1]
```diff
@@ -49,6 +49,8 @@
         name='hkm_ajax_record_fav'),
     url(r'^ajax/crop/$', views.AjaxCropRecordView.as_view(),
         name='hkm_ajax_crop'),
+        url(r'^ajax/collection/$', views.AjaxAddToCollection.as_view(),
+            name='hkm_add_to_collection'),
```
Comment (tuomas777) [inline hkm/urls.py:53]: Indentation seems to be a bit off.

File: hkm/views/views.py [hunk 2]
```diff
@@ -1096,23 +1091,34 @@ def handle_download(self, request, *args, **kwargs):
...
+class AjaxAddToCollection(View):
+    name = 'hkm_add_to_collection'
+
+    def post(self, request, *args, **kwargs):
+        action = request.POST['action']
+        if action == 'add':
+            return self.handle_add_to_collection(request, *args, **kwargs)
+        elif action == 'add-create-collection':
+            return self.handle_add_to_new_collection(request, *args, **kwargs)
+        return http.HttpResponseBadRequest()
```
Comment (tuomas777) [inline hkm/views/views.py:1104]: Is the requirement of an authenticated user for these actions removed intentionally?

Comment (tuovinensanttu) [reply hkm/views/views.py:1104]: The button is hidden in the frontend if user is not authenticated, so I thought that would be enough. Should I add it to backend as well?

File: hkm/urls.py (hunk 1 above)
Comment (tuovinensanttu) [reply hkm/urls.py:53]: Uups, I'll fix that.

File: hkm/views/views.py [hunk 3]
```diff
@@ -1096,23 +1091,35 @@ def handle_download(self, request, *args, **kwargs):
...
+    name = 'hkm_add_to_collection'
+
+    def post(self, request, *args, **kwargs):
+        if request.user.is_authenticated():
+            action = request.POST['action']
+            if action == 'add':
+                return self.handle_add_to_collection(request, *args, **kwargs)
+            elif action == 'add-create-collection':
+                return self.handle_add_to_new_collection(request, *args, **kwargs)
+        return http.HttpResponseForbidden()
```
Comment (tuomas777) [inline hkm/views/views.py:1105]: Now also incorrect actions will get this 403 which is a bit off. Not a huge deal in this exact spot ofc.

File: hkm/views/views.py [hunk 4]
```diff
@@ -1096,23 +1091,35 @@ def handle_download(self, request, *args, **kwargs):
                   'data': {'url': tmp_image.edited_image.url}})
         return http.JsonResponse({'url': tmp_image.edited_image.url})
 
+
+class AjaxAddToCollection(View):
+    name = 'hkm_add_to_collection'
```
[the remark this answers is not in the document]
Comment (tuovinensanttu) [reply hkm/views/views.py]: Removed.

File: hkm/views/views.py (hunk 3 above)
Comment (tuovinensanttu) [reply hkm/views/views.py]: Changed this back to `http.HttpResponseBadRequest()`

**Review by tuomas777 — APPROVED** [review]
LGTM 👍

**Pull request merged**